tests/app/service/test_service_contact_list_rest.py

```diff
@@ -113,3 +129,77 @@ def test_dao_get_contact_list_by_id_does_not_return_if_contact_list_id_for_anoth
     )
 
     assert response['message'] == "No result found"
+
+
+def test_archive_contact_list_by_id(mocker, admin_request, sample_service):
+    mock_s3 = mocker.patch('app.service.rest.s3.remove_contact_list_from_s3')
+    service_1 = create_service(service_name='Service under test')
+    template_1 = create_template(service=service_1)
+    expected_list = create_service_contact_list(service=service_1)
+    other_list = create_service_contact_list(service=service_1)
+
+    job_1 = create_job(template=template_1, contact_list_id=expected_list.id)
```

Comment:
Optional: Are all the jobs so this test is not just testing that you archive a contact list by ID but also testing that you can archive a contact list which has already been linked to a job? If so, would be nice to make that explicit with either a comment or splitting those into two tests as they are testing two different things

Reply:
Comment added in 14605764